pyfar/dsp/dsp.py

```diff
@@ -1487,4 +1494,363 @@ def decibel(signal, domain='freq', log_prefix=None, log_reference=1):
+        shape of ``signal.time`` and ``signal.freq``.
+        The default is ``False``.
+
+    Returns
+    -------
+    data : numpy.ndarray
+        The RMS of the input signal.
+
+    Notes
+    -----
+    The RMS equals the square root of the signal's power.
+    :py:func:`~pyfar.dsp.energy` and :py:func:`~pyfar.dsp.power` can be used
+    to compute the energy and the power.
+    """
+    # check input
+    if not isinstance(signal, pyfar.Signal):
+        raise ValueError(f"signal is type '{signal.__class__}'"
+                         " but must be of type 'Signal'.")
+    # cumpute rms data
+    data = np.sqrt(np.sum(signal.time**2, axis=-1)/signal.n_samples)
+    if keepdims:
+        data = data[..., None]
+    return data
+
+
+def normalize(signal, reference_method='max', domain='time',
+              channel_handling='each', target=1, limits=(None, None),
+              unit=None, return_reference=False):
+    """
+    Apply a normalization.
+
+    In the default case, the normalization ensures that the maximum absolute
+    amplitude of the signal after normalization is 1. This is achieved by
+    the multiplication
+
+    ``signal_normalized = signal * target / reference``,
+
+    where `target` equals 1 and `reference` is the maximum absolute amplitude
+    before the normalization.
+
+    Several normalizations are possible, which in fact are different ways
+    of computing the `reference` value (e.g. based on the spectrum).
+    See the parameters for details.
+
+    Parameters
+    ----------
+    signal: Signal, TimeData, FrequencyData
+        Input signal.
+    domain: string
+        Determines which data is used to compute the `reference` value.
+
+        ``'time'``
+           Use the absolute of the time domain data ``np.abs(signal.time)``.
+        ``'freq'``
+          Use the magnitude spectrum `np.abs(`signal.freq)``. Note that the
+          normalized magnitude spectrum used
+          (cf.:py:mod:`FFT concepts <pyfar._concepts.fft>`).
+
+        The default is ``'time'``.
+    reference_method: string, optional
+        Reference method to compute the channel-wise `reference` value using
+        the data according to `domain`.
+
+        ``'max'``
+            Compute the maximum absolute value per channel.
+        ``'mean'``
+            Compute the mean absolute values per channel.
+        ``'energy'``
+            Compute the energy per channel using :py:func:`~pyfar.dsp.energy`.
+        ``'power'``
+            Compute the power per channel using :py:func:`~pyfar.dsp.power`.
+        ``'rms'``
+            Compute the RMS per channel using :py:func:`~pyfar.dsp.rms`.
+
+        The default is ``'max'``.
+    channel_handling: string, optional
+        Define how channel-wise `reference` values are handeled for multi-
+        channel signals. This parameter does not affect single-channel signals.
+
+        ``'each'``
+            Separate normalization of each individual channel.
```

Comment:
```suggestion
        ``'individual'``
            Separate normalization of each channel individually.
```
"each" implies, that each channel is handled in some way, but does not give information on how each channel is handled. In all of the stated cases, each channel is considered for normalizing, but each alone does not state how the information of each channel is used.